test/test_serial.py

```diff
@@ -70,11 +72,13 @@ def test_basic(self):
     def test_jobs(self):
         '''The jobs method yields None if there are no jobs'''
         worker = NoListenWorker(['foo'], self.client, interval=0.2)
-        self.assertEqual(worker.jobs().next(), None)
+        for job in worker.jobs():
```

Comment:
Can't this be:
```python
self.assertEqual(next(worker.jobs()), None)
```
I think that more closely matches the original test and is similar to the changes you do down below.